scripts/globals/utils.lua

```diff
@@ -43,6 +43,22 @@ function utils.bind(func, ...)
     end
 end
 
+-- Creates a slice of an input table and returns a new table
+function utils.slice(inputTable, first, last, step)
+    local slicedTable = {}
+    first = first or 1
+    last = last or #inputTable
+    step = step or 1
+    local position = 1
+
+    for i = first, last, step do
+        slicedTable[position] = inputTable[i]
```

Comment:
Is it possible for this to step over the end and place a nil in the new table?

Reply:
Not from what I tested. If step > 1 then it will properly stop not place a nil. Even if it did though it wouldn't cause problems since it'd be nil.